still some warnings with -Wswitch-default

Hello (again),

The following piece of code (see below) is ok under
flex (2.6.1 or 2.6.2) but still produces annoying warnings
when given to the -Wswitch-default option of gcc (i'm using gcc-6.2.0):

cm2.cc: In function 'int yylex()':
cm2.cc:814:15: warning: switch missing default case [-Wswitch-default]

cm2.cc: In function 'int yyinput()':
cm2.cc:1140:11: warning: switch missing default case [-Wswitch-default]

Addition of a simple 'default: break;' at various places inside the
flex code (2.6.1 or 2.6.2) corrects the problem.

%{
%}
%S INSIDE
%%
"beg of inside" { ECHO; BEGIN INSIDE; };
<INSIDE>"end of inside" { ECHO; BEGIN INITIAL; };
%%

--- a/src/c99-flex.skl
+++ b/src/c99-flex.skl
@@ -1214,6 +1214,8 @@ int yyinput (yyscan_t yyscanner)
 			case EOB_ACT_CONTINUE_SCAN:
 				yyscanner->yy_c_buf_p = yyscanner->yytext_ptr + offset;
 				break;
+			default:
+				yypanic("unexpected return value from yy_get_next_buffer()", yyscanner);
 			}
 		}
 	}
@@ -2363,6 +2365,8 @@ m4_ifdef([[M4_MODE_FIND_ACTION_REJECT_OR_INTERACTIVE]], [[
 						yy_cp = yyscanner->yy_c_buf_p;
 						yy_bp = yyscanner->yytext_ptr + YY_MORE_ADJ;
 						goto yy_find_action;
+					default:
+						yypanic("unexpected return value from yy_get_next_buffer()", yyscanner);
 					} /* end EOB inner switch */
 				} /* end if */
 				break;

--- a/src/cpp-flex.skl
+++ b/src/cpp-flex.skl
@@ -2269,6 +2269,8 @@ m4_ifdef([[M4_MODE_FIND_ACTION_REJECT_OR_INTERACTIVE]], [[
 						yy_cp = YY_G(yy_c_buf_p);
 						yy_bp = YY_G(yytext_ptr) + YY_MORE_ADJ;
 						goto yy_find_action;
+					default:
+						YY_FATAL_ERROR("unexpected return value from yy_get_next_buffer()");
 					} /* end EOB inner switch */
 				} /* end if */
 				break;
@@ -2789,6 +2791,8 @@ int yyFlexLexer::yyinput()
 			case EOB_ACT_CONTINUE_SCAN:
 				YY_G(yy_c_buf_p) = YY_G(yytext_ptr) + offset;
 				break;
+			default:
+				YY_FATAL_ERROR("unexpected return value from yy_get_next_buffer()");
 			}
 		}
 	}

--- a/src/go-flex.skl
+++ b/src/go-flex.skl
@@ -1123,6 +1123,8 @@ int yyinput(FlexLexer *yyscanner)
 			case EOB_ACT_CONTINUE_SCAN:
 				yyscanner->yyCBufP = yyscanner->yytext_ptr + offset;
 				break;
+			default:
+				yypanic("unexpected return value from yy_get_next_buffer()", yyscanner);
 			}
 		}
 	}
@@ -2213,6 +2215,8 @@ m4_ifdef([[M4_MODE_FIND_ACTION_REJECT_OR_INTERACTIVE]], [[
 						yyCp = yyscanner->yyCBufP;
 						yyBp = yyscanner->yytext_ptr + YY_MORE_ADJ;
 						goto yyFindActionLabel;
+					default:
+						yypanic("unexpected return value from yy_get_next_buffer()", yyscanner);
 					} /* end EOB inner switch */
 				} /* end if */
 				break;

